draft-ietf-quic-tls.md

```diff
@@ -499,12 +499,23 @@ handshake, new data is requested from TLS after providing received data.
 ### Encryption Level Changes
 
 As keys at a given encryption level become available to TLS, TLS indicates to
-QUIC that reading or writing keys at that encryption level are available.  While
-generating these keys, an endpoint SHOULD buffer received packets marked as
-protected by the keys being generated, and process them once those keys become
-available.  If the keys are generated asynchronously, an endpoint MAY continue
-responding to the received packets that were processable while waiting for TLS
-to provide these keys.
+QUIC that reading or writing keys at that encryption level are available.
+
+The events that cause new keys to be available are not asynchronous; they
```

Comment:
Given that you describe in the following paragraph that the keys might well results from asynchronous processing, I'm not convinced that "not asynchronous" is the proper claim to make here.  I think this claim is really trying to say that key generation isn't spontaneous, and you don't have to have a callback mechanism for TLS to just decide to give you keys out of the blue.  However (following paragraph), the time from input to completion-with-keys can take a while and shouldn't be blocking.

Perhaps something like "The availability of new keys is always a result of inputs provided to TLS."?

Reply:
I think dropping "asynchronous" from this sentence is fine -- the next para holds up well in describing what asynchronous actually means.